File: CPP/Sorting/Divide Players Into Teams of Equal Skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        int n =skill.size();
        sort(skill.begin(),skill.end());
        long long chemistry=0;
        int totalSkill = skill[0]+skill[n-1];
        for(int i=0;i<n/2;i++){
            int currSkill = skill[i]+skill[n-i-1];
            if(totalSkill!=currSkill){
                return -1;
            }
            chemistry+=skill[i]*skill[n-i-1];
        }
        return chemistry;
    }
};

File: Python/Sorting/Divide Players Into Teams of Equal Skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        n = len(skill)
        skill.sort()
        chemistry = 0
        totalSkill = skill[0]+skill[n-1]
        i = 0
        while i<n/2:
            currSkill = skill[i]+skill[n-1-i]
            if currSkill!=totalSkill:
                return -1
            chemistry+=skill[i]*skill[n-i-1]
            i+=1

        return chemistry


       
